src/mbgl/map/map_context.cpp

```diff
@@ -302,9 +306,16 @@ void MapContext::renderStill(StillImageCallback fn) {
     }
 
     callback = fn;
+    mayRender = true;
     triggerUpdate(Update::RenderStill);
 }
 
+void MapContext::renderSync() {
+    mayRender = true;
```

Comment:
Since `render` doesn't check `mayRender`, it seems like it's equivalent to "is a still render pending?". Can we use `callback` in `update()` instead, like:

```
if (callback) {
    render();
} else {
    view.invalidate();
}
```

Or is there something I'm missing?